firewood/src/hashednode.rs

```diff
@@ -111,69 +96,81 @@ impl<T: ReadLinearStore> HashedNodeStore<T> {
 }
 
 impl<T: WriteLinearStore> HashedNodeStore<T> {
-    // recursively hash this node
+    // Hashes the node at `node_addr` and all of its children recursively.
+    // Returns the hash of the node and the address of the node.
     fn hash(
         &mut self,
         node_addr: LinearAddress,
-        node: &mut Node,
         path_prefix: &mut Path,
-    ) -> Result<TrieHash, Error> {
-        match node {
-            Node::Branch(b) => {
-                // We found a branch, so find all the children that need hashing
-                let mut modified = false;
+    ) -> Result<(TrieHash, LinearAddress), Error> {
+        let modified = self.added.remove(&node_addr);
```

Comment:
nit: I think this was easier to read with the external function that isolated the magic about into_inner.